skripchenkoveta/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+EXIT_COMMAND = 'exit!'
+
+def data_extraction
+  file = File.new('data.txt', 'r:UTF-8')
+  file.read
+end
+
+def town_search(town, str)
+  str = str.split("\n")
+  str.each do |i|
+    next unless i.include? town
+
+    return i
+  end
+end
+
+def extracting_numbers(str)
+  str.scan(/\d*\.\d/).map(&:to_f)
+end
+
+def calculate_middle_rainfall(town, str)
+  avg = 0.0
```

Comment:
do not introduce new variable until you are sure it will be used during the method execution. It's possible method execution stops on the next line :)